ledger-tool/src/main.rs

```diff
@@ -2980,16 +2980,24 @@ fn main() {
                     value_t_or_exit!(arg_matches, "snapshot_slot", Slot)
                 };
 
-                assert!(
-                    blockstore.meta(snapshot_slot).unwrap().is_some(),
-                    "snapshot slot doesn't exist"
-                );
+                if blockstore
+                    .meta(snapshot_slot)
+                    .unwrap()
+                    .filter(|m| m.is_full())
```

Comment:
Huh, didn't realize you could do `filter()` on an `Option`; think I've always done something like:
```rust
.meta(snapshot_slot)
.map(|meta| !meta.is_full())
.unwrap_or_else(true)
```
What you have is good, just making the today-I-learned comment